homeassistant/components/plex/websockets.py

```diff
@@ -0,0 +1,74 @@
+"""Support for Plex websockets."""
+import asyncio
+import logging
+
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .const import PLEX_UPDATE_PLATFORMS_SIGNAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WebsocketPlayer:
+    """Represent a player in the Plex websocket stream."""
+
+    def __init__(self, state, media_key, position):
+        """Initialize a WebsocketPlayer instance."""
+        self.state = state
+        self.media_key = media_key
+        self.position = position
+
+
+async def websocket_handler(hass, server_id, uri):
+    """Create websocket connection thread and handle received messages."""
+    websocket_players = {}
+
+    session = async_get_clientsession(hass)
+
+    while True:
+        async with session.ws_connect(uri) as ws_client:
```

Comment:
This, the websocket connection and message parsing, should be handled by a standalone library.